scripts/irods/upgrade_configuration.py

```diff
@@ -345,6 +347,41 @@ def update_base(base, updates):
     if "irods_tcp_keepalive_time_in_seconds" in service_account_environment:
         new_server_config["tcp_keepalive_time_in_seconds"] = service_account_environment.get("irods_tcp_keepalive_time_in_seconds", None)
 
+    # Update database plugin config (if present) to new format
+    if 'plugin_configuration' in new_server_config and 'database' in new_server_config['plugin_configuration']:
+        # Only focus on part we care about
+        database_config_options = new_server_config['plugin_configuration']['database']
+
+        if len(database_config_options) != 0:
```

Comment:
Could reduce indentation by one for the rest of this block with...
```python
        if len(database_config_options) == 0:
            return new_server_config
```
Honestly, if it's working, you should probably ignore this. Just something I noticed and figured I'd throw it out there.

Reply:
The only reason I didn't do this is that if someone were to append some code, or shuffle the function around, it might skip some things unintentionally.

Reply:
Fair enough. Let's leave it how it is.